src/dex/clear/clear.ts

```diff
@@ -0,0 +1,437 @@
+import { Interface } from '@ethersproject/abi';
+import { DeepReadonly } from 'ts-essentials';
+import { Log, Logger, Token, Address } from '../../types';
+import { IDexHelper } from '../../dex-helper/idex-helper';
+import {
+  PoolPrices,
+  AdapterExchangeParam,
+  SimpleExchangeParam,
+  PoolLiquidity,
+  ExchangePrices,
+} from '../../types';
+import { IDex } from '../idex';
+import { SimpleExchange } from '../simple-exchange';
+import { ClearConfig, Adapters } from './config';
+import { ClearData, DexParams } from './types';
+import { Network, NULL_ADDRESS, SwapSide } from '../../constants';
+import * as CALLDATA_GAS_COST from '../../calldata-gas-cost';
+import { getDexKeysWithNetwork } from '../../utils';
+import clearSwapAbi from '../../abi/clear/ClearSwap.json';
+import clearFactoryAbi from '../../abi/clear/ClearFactory.json';
+import { NumberAsString } from '@paraswap/core';
+
+const CLEAR_GAS_COST = 150_000;
+
+interface ClearVault {
+  id: string;
+  address: string;
+  tokens: ClearVaultToken[];
+}
+
+interface ClearVaultToken {
+  id: string;
+  address: string;
+  symbol: string;
+  decimals: string;
+}
+
+/**
+ * Clear DEX Integration for ParaSwap
+ *
+ * Clear is a custom multi-token vault protocol (NOT a standard AMM)
+ * - Vaults contain N tokens (not just 2 like Uniswap pools)
+ * - Pricing via ClearSwap.previewSwap() RPC call (no x*y=k formula)
+ * - Discovery via GraphQL indexer
+ * - Each (vault, tokenA, tokenB) combination = one "pool" for ParaSwap
+ */
+export class Clear extends SimpleExchange implements IDex<ClearData> {
+  static dexKeys = ['clear'];
+  logger: Logger;
+  protected config: DexParams;
+  protected adapters: { [side: string]: { name: string; index: number }[] };
+
+  readonly hasConstantPriceLargeAmounts = false;
+  readonly isFeeOnTransferSupported = false;
+
+  public static dexKeysWithNetwork: { key: string; networks: Network[] }[] =
+    getDexKeysWithNetwork(ClearConfig);
+
+  readonly clearSwapIface = new Interface(clearSwapAbi);
+  readonly clearFactoryIface = new Interface(clearFactoryAbi);
+
+  // Cache for vaults to reduce GraphQL calls
+  private vaultsCache: Map<string, ClearVault[]> = new Map();
+  private vaultsCacheTimestamp: Map<string, number> = new Map();
+  private readonly CACHE_TTL = 60 * 1000; // 1 minute
+
+  constructor(
+    readonly network: Network,
+    readonly dexKey: string,
+    readonly dexHelper: IDexHelper,
+    adapters?: { [side: string]: { name: string; index: number }[] },
```

Comment:
side should be of the SwapSide don't you think ?

Also, shouldn't the whole type of Adapters be AdaptersConfig instead of a type declared on the flight ??